apps/api/src/app/events/usecases/send-message/send-message.usecase.ts

```diff
@@ -174,11 +175,15 @@ export class SendMessage {
       environmentId: job._environmentId,
     });
 
+    const subscriber = await this.subscriberRepository.findById(job._subscriberId);
+    if (!subscriber) throw new ApiException('Subscriber not found with id ' + job._subscriberId);
+
     const buildCommand = GetSubscriberTemplatePreferenceCommand.create({
       organizationId: job._organizationId,
-      subscriberId: job._subscriberId,
+      subscriberId: subscriber.subscriberId,
```

Comment:
to make sure we only pass here the public subscriberId to avoid confusion

Reply:
that one will help, I refactored the job entity in https://github.com/novuhq/novu/pull/2489
maybe I will look into finalizing it on the cool down in order to merge it as well.

Reply:
This flow is too confusing! that's why I refactored it on the #2489 😀